demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/batches/BatchesActivity.java

```diff
@@ -0,0 +1,122 @@
+package com.novoda.downloadmanager.demo.extended.batches;
+
+import android.database.ContentObserver;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.ListView;
+
+import com.novoda.downloadmanager.DownloadManagerBuilder;
+import com.novoda.downloadmanager.demo.R;
+import com.novoda.downloadmanager.demo.extended.Download;
+import com.novoda.downloadmanager.demo.extended.DownloadAdapter;
+import com.novoda.downloadmanager.demo.extended.QueryForDownloadsAsyncTask;
+import com.novoda.downloadmanager.demo.extended.QueryTimestamp;
+import com.novoda.downloadmanager.lib.DownloadManager;
+import com.novoda.downloadmanager.lib.NotificationVisibility;
+import com.novoda.downloadmanager.lib.Query;
+import com.novoda.downloadmanager.lib.Request;
+import com.novoda.downloadmanager.lib.RequestBatch;
+import com.novoda.notils.logger.simple.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchesActivity extends AppCompatActivity implements QueryForDownloadsAsyncTask.Callback {
+    private static final String BIG_FILE = "http://download.thinkbroadband.com/100MB.zip";
+    private static final String BEARD_IMAGE = "http://i.imgur.com/9JL2QVl.jpg";
+
+    private final Handler handler = new Handler(Looper.getMainLooper());
+    private DownloadManager downloadManager;
+    private ListView listView;
+    private DownloadAdapter downloadAdapter;
+
+    private final QueryTimestamp lastQueryTimestamp = new QueryTimestamp();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_batches);
+        listView = (ListView) findViewById(R.id.main_downloads_list);
+        downloadManager = DownloadManagerBuilder.from(this)
+                .withVerboseLogging()
+                .build();
+        downloadAdapter = new DownloadAdapter(new ArrayList<Download>());
```

Comment:
i think in the future we should have multiple adapters per package as well so that it's easily understandable what each demo does on it's own, but for now :+1: